### DISCUSSION TITLE
Key Mapping to handle Escape Characters

### DISCUSSION THREAD

**Discussion by nomad3k** [helix-editor/helix#9074] [Troubleshooting]
A big issue for me with helix is the inability to use the Command key on my Mac.  I appreciate that this is because of an upstream issue with other tools, but a solution exists for other software.

iTerm2 allows me to setup a "Natural Text Editing" key mapping, which maps the following:
  ⌘← = 0x01 = ^A
  ⌘→ = 0x05 = ^E
  ⌥← = 0x02 = ^B
  ⌥→ = 0x06 = ^F

These all function as expected (start of line, end of line, start of prev word, and start of next work respectively) in the console.  In helix though they insert "a", "e", "b" & "f".

Therefore, could we amend the key mapping function to support them?

**Comment by the-mikedavis** [comment]
I assume that iterm means <kbd>Ctrl</kbd>+<kbd>a</kbd> by `^A`. You should be able to map those sequences, for example:

```toml
# ~/.config/helix/config.toml
[keys.insert]
C-a = "goto_line_start"
```
